From c48e7df4bb92c765abea8211b959da4f2a452666 Mon Sep 17 00:00:00 2001
From: Sebastian Krzyszkowiak <dos@dosowisko.net>
Date: Wed, 20 Feb 2019 03:37:27 +0100
Subject: [PATCH 5/6] wayland: add dummy remove handler to wl_registry_listener

Passing NULL there causes SDL to crash every time the compositor
sends a remove event - which is particularly unhelpful as the protocol
allows to simply ignore the event, because removed objects can be still
safely called on the client side and compositor will ignore such requests.
---
 src/video/wayland/SDL_waylandvideo.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/video/wayland/SDL_waylandvideo.c b/src/video/wayland/SDL_waylandvideo.c
index a1849d8d1..ab4fde0e6 100644
--- a/src/video/wayland/SDL_waylandvideo.c
+++ b/src/video/wayland/SDL_waylandvideo.c
@@ -392,9 +392,12 @@ display_handle_global(void *data, struct wl_registry *registry, uint32_t id,
     }
 }
 
+static void
+display_remove_global(void *data, struct wl_registry *registry, uint32_t id) {}
+
 static const struct wl_registry_listener registry_listener = {
     display_handle_global,
-    NULL, /* global_remove */
+    display_remove_global
 };
 
 int
-- 
2.20.1

